src/osd/OSD.cc

```diff
@@ -3251,6 +3251,8 @@ struct pistate {
 void OSD::build_past_intervals_parallel()
 {
   map<PG*,pistate> pis;
+  set<PG*> fixed;
+  OSDMapRef cur_map, last_map;
```

Comment:
could also move these definitions to https://github.com/ceph/ceph/pull/8390/commits/657c7f01b99e08f10bc8d613c20a0870df619cbc#diff-fa6c2eba8356ae1442d1bf749beacfdfR3287, closer to where the defined variables are used. but i'd suggest reuse the logic here after confirmed by @dzafman .